tcf_website/models/models.py

```diff
@@ -620,14 +620,19 @@ def average(self):
         return (self.instructor_rating +
                 self.recommendability + self.enjoyability) / 3
 
-    # not sure if this gets used anywhere either
     def count_votes(self):
         """Sum votes for review."""
-        vote_sum = self.vote_set.aggregate(
-            models.Sum('value')).get('value__sum', 0)
-        if not vote_sum:
-            return 0
-        return vote_sum
+        upvotes = self.vote_set.filter(value=1)\
+            .aggregate(models.Sum('value'))\
+            .get('value__sum', 0)
+        downvotes = self.vote_set.filter(value=-1)\
+            .aggregate(models.Sum('value'))\
+            .get('value__sum', 0)
+
+        upvotes = upvotes if upvotes else 0
+        downvotes = -1 * downvotes if downvotes else 0
+
+        return {'upvotes': upvotes, 'downvotes': downvotes}
```

Comment:
The following code should be more efficient because it has one database query and no additional Python computations, although less readable. Note that `Coalesce` and `Abs` are in `django.db.models.functions`:
```suggestion
        return self.vote_set.aggregate(
            upvotes=Coalesce(Sum('value', filter=Q(value=1)), 0),
            downvotes=Coalesce(Abs(Sum('value', filter=Q(value=-1))), 0),
        )
```
There is a tradeoff between performance and readability, so feel free to dismiss this comment.

Reply:
Cool, I went ahead and made those changes.